### PR TITLE
bq query optimizations (consumer side)

### PR DESCRIPTION
**Pull request by gkroiz**
# Description
Consumer side BQ queries optimizations for xl ml dashboard. These changes were made to improve query time while maintaining/improving readability. More details [here](https://tinyurl.com/2979uh88). 

# Tests
- Tests within `dashboard/` passed.
- To check that the logic of the new queries matches the old queries, I tested the queries against each other using different test names, metric names and over different periods of time. From each of these comparisons, the number of entries matched, the queried rows matched, and the ordering matched when ordering mattered.
- Ran dashboard locally and tested functionality for heatmap, metrics plotting, and metrics compare

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run one-shot tests and provided workload links above if applicable. 
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by RissyRan — COMMENTED** [review]
Thanks Gerson for the change. Overall LGTM, just 2 minor comments:

1) Could you add a brief summary of results to `Compared results from each query to old queries in BigQuery for several test_name/metrics configurations` in the PR description. No screenshots needed.

2) Could you ensure the format of query is consistent? I see left parenthesis is not, i.e. line#59 vs. line#45 in `main_heatmap.py` file. Please also check others.

**Comment by gkroiz** [comment]
Thank you @RissyRan for the feedback. I added a bit more to the PR description. Please let me know if this should be more descriptive. Also made adjustments to standardize formatting.

File: dashboard/main_heatmap.py [hunk 1]
```diff
@@ -34,34 +34,42 @@
...
+SELECT 
+  test_name,
+  job_status,
+  SAFE_CAST(DATE(timestamp, 'US/Pacific') AS STRING) as run_date,
+  logs_link,
+  workload_link,
+  uuid
 FROM (
-  SELECT
+  SELECT
```
Comment (RissyRan) [inline dashboard/main_heatmap.py:45]: Nit: extra space after the SELECT?

Comment (gkroiz) [reply dashboard/main_heatmap.py]: Fixed

**Review by RissyRan — APPROVED** [review]

**Pull request merged**